### PR TITLE
FossGraph. Added vulnerability removal

### LINKED ISSUE #2036: FossGraph. Add vulnerability removal
Need add vulnerability removal

### PR DESCRIPTION
**Pull request by Cheshiriks**
* added vulnerability removal
* added user in vulnerability

Closes #2036

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt [hunk 1]
```diff
@@ -54,14 +58,21 @@ class VulnerabilityService(
...
+     * @param authentication auth info of a current user
      */
     @Transactional
-    fun save(vulnerabilityDto: VulnerabilityDto) {
+    fun save(
+        vulnerabilityDto: VulnerabilityDto,
+        authentication: Authentication,
+    ) {
+        val userId = (authentication.details as AuthenticationDetails).id
+        val user = userRepository.findById(userId).get()
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:69]:
```suggestion
import com.saveourtool.save.utils.getByIdOrNotFound
...
        val user = userRepository.getByIdOrNotFound(userId)
```

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 2]
```diff
@@ -75,10 +75,12 @@ class VulnerabilityController(
         description = "Save vulnerability.",
     )
     @ApiResponse(responseCode = "200", description = "Successfully saved vulnerability")
+    @PreAuthorize("permitAll()")
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:78]: is it a temporary solution? or everyone can save it?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt [hunk 3]
```diff
@@ -111,6 +113,15 @@ val fossGraph: FC<FossGraphViewProps> = FC { props ->
             }
 
         setVulnerability(vulnerabilityNew)
+
+        val userInfo = get(
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt:117]:
```suggestion
        val userInfo: UserInfo = get(
```
and remove suppressing

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:78]: Yes. Anyone logging in can create a new vulnerability.

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt]: Fixed

**Review by nulls — APPROVED** [review]

**Pull request closed without merging**